### PR TITLE
Updated Rebellion Skills

### LINKED ISSUE #1449: Rebellion skill Binding trap does 1 damage when using  Flicker
I think the title pretty much explains it. I'm using rAthena latest version. 

Steps to reproduce: Use binding trap, someone else steps on it, use Flicker and it does 1 HP damage. Also it slows down the caster (but i don't know if this is the correct behavior or not) and it seems like you can't cast on the same cell of someone else or yourself.

### PR DESCRIPTION
**Pull request by cydh**
* Updated damage formula
* Updated effect formula
* Fixed #1449
* References: iRO Wiki pages about Rebellion Skill, accessed today (Aug 3, 2017)

Signed-off-by: Cydh Ramdh <cydh@pservero.com>

### CODE REVIEW

File: src/map/battle.c [hunk 1]
```diff
@@ -2926,6 +2926,11 @@ static struct Damage battle_calc_attack_masteries(struct Damage wd, struct block
 				ATK_ADD(wd.masteryAtk, wd.masteryAtk2, 10 * sc->data[SC_GN_CARTBOOST]->val1);
 #endif
 			}
+			if (sc->data[SC_P_ALTER])
+				ATK_ADD(wd.damage, wd.damage2, sc->data[SC_P_ALTER]->val2);
+#ifdef RENEWAL
+				ATK_ADD(wd.masteryAtk, wd.masteryAtk2, sc->data[SC_P_ALTER]->val2);
+#endif
 		}
```
Comment (cydh) [inline src/map/battle.c]: forgotten `{ ... }`

File: src/map/battle.c [hunk 2]
```diff
@@ -4189,52 +4189,62 @@ static int battle_calc_attack_skill_ratio(struct Damage wd, struct block_list *s
...
 			skillratio += -100 + (2500 + 500 * skill_lv);
 			break;
 		case RL_R_TRIP:
-			skillratio += -100 + 150 * skill_lv; //(custom)
-			break;
 		case RL_R_TRIP_PLUSATK:
-			skillratio += -100 + 100 * skill_lv + rnd()%10 + 100; //(custom)
+			// iRO Wiki: [(DEX � 2) � (10 + (Skill_Lv � 3)]%
+			// Collision damage is half the initial damage
+			skillratio += -100 + ((sstatus->dex / 2) * (10 + (skill_lv * 3)) * (skill_id == RL_R_TRIP_PLUSATK) ? 0.5 : 1);
```
Comment (cydh) [inline src/map/battle.c]: parentheses

File: src/map/battle.c [hunk 3]
```diff
@@ -4189,52 +4189,62 @@ static int battle_calc_attack_skill_ratio(struct Damage wd, struct block_list *s
...
+			// Base_Damage: [1600 + 800 * Skill_Level]
+			skillratio += -100 + (1600 + (800 * skill_lv)) + (200 * (sd ? sd->spiritball_old+1 : 2) / 2);
 			break;
 		case RL_QD_SHOT:
 			skillratio += -100 + (max(pc_checkskill(sd,GS_CHAINACTION),1) * status_get_dex(src) / 5); //(custom)
 			break;
 		case RL_FIRE_RAIN:
-			skillratio += -100 + 2000 + (200 * (skill_lv - 1)) + status_get_dex(src); //(custom) //kRO Update 2013-07-24. 2,000% + caster's DEX (?) [Cydh]
+			// iRO Wiki: [{2000 + (Skill_Lv � DEX)} � Base_Level � 100]%
+			skillratio += -100 + (2000 * sstatus->dex);
```
Comment (cydh) [inline src/map/battle.c]: what the -_-
``` skillratio += -100 + (2000 + skill_lv * sstatus->dex); ```

**Comment by aleos89** [comment]
Aug 3, 2017!? A time traveler!? I would like the Mega Million lotto ticket winning numbers a year from now, please!

I have started on a large rewrite of Rebellion skills as there were tons of changes, not just damage adjustments on kRO.

**Comment by zackdreaver** [comment]
@aleos89 
> I have started on a large rewrite of Rebellion skills as there were tons of changes

A new rebel rebalance patch, which came last month, or just fixing the formula, based on current one in iRO?

**Comment by aleos89** [comment]
It's based off the one from May 2016.

**Comment by zackdreaver** [comment]
@aleos89 
nice!

**Comment by cydh** [comment]
OK

**Pull request closed without merging**